builtin/credential/approle/path_role.go

```diff
@@ -867,6 +883,18 @@ func (b *backend) pathRoleRead(req *logical.Request, data *framework.FieldData)
 			resp.AddWarning("Role does not have any constraints set on it. Updates to this role will require a constraint to be set")
 		}
 
+		// People have been complaining about role becoming useless despite
+		// being able to read it. For sanity, verify that the index from
+		// role_id to role_name still exists. If the index is missing, return a
+		// warning.
+		roleIDIndex, err := b.roleIDEntry(req.Storage, role.ID)
+		if err != nil {
+			return nil, err
+		}
+		if roleIDIndex == nil {
```

Comment:
Wouldn't the right thing to do here be to instead ensure that it's created?

Reply:
I thought of that too and considered that to be “magic”. It’s would be a cover up and may not lead us to the actual problem.

Reply:
I thought of that too and considered that to be “magic”. It’s would be a cover up and may not lead us to the actual problem.